### ISSUE TITLE
EmEditor syntax.esy and macro.jsee text files are not seen as text by @github

### ISSUE THREAD

**Issue by jesus2099** [jesus2099/settings#2]
Example 807bc1f1077d54be35df2be0f1b729155b60a2f1 where it says _**Binary file not shown.**_ instead of showing text diff (files.jsee are JavaScript text files).

**Comment by jesus2099** [comment]
I made a post in github.community about it in 2017: https://github.community/t5/How-to-use-Git-and-GitHub/files-jsee-JavaScript-but-Binary-file-not-shown/m-p/608

I noticed someone else with same question in 2019: https://github.community/t5/GitHub-Pages/Source-files-shown-as-binary-files/m-p/35456

The answer says _Try creating a `.gitattributes` file_ with `*.jsee text` But I'm not sure if it will force github to not see files.jsee as being binaries and showing diff for them. I thought `.gitattributes` was more related to git command new line management.

**Comment by jesus2099** [comment]
Same for `.esy` configuration files in plain text: https://github.com/jesus2099/settei/commit/b0c4692486041ba7b20daaa1f40c0a5318f96ff1 .

**Comment by jesus2099** [comment]
It does not work:

> Binary file not shown.

**Comment by jesus2099** [comment]
😱 **Wow it's even worse! Git corrupts UTF-16 files, completely!**

**Comment by jesus2099** [comment]
WOW! Here is maybe why my UTF-16 LE files were corrupted by Git (I've been using both Windows and Linux on my repo):

> https://github.com/git/git/pull/1294/commits
> https://github.com/git/git/pull/1294/commits/4d0043712d32f95cda5bd28efbbf63408ec93ce3
>
> @mbs-c:
>
> _Perforce silently replaces LF with CR LF for "utf16" files if the client is a native Windows client. Since git's autocrlf logic does not undo this transformation for UTF-16 encoded files, **git-p4 replaces CR LF with LF during the sync if the file type "utf16" is detected** and the Perforce client platform indicates that this conversion is performed._
>
> _Windows only runs on little-endian architectures, therefore the encoding of the byte stream received from the Perforce client is UTF-16-LE and the relevant byte sequence is `0D 00 0A 00`._

(I don't know what is _Perforce_, though)

But I have of course deactivated `autocrlf` in all my repos, since the the beginnings, so I'm not sure I understand.

**Comment by mbs-c** [comment]
If you've never used Perforce, you've never used git-p4 and my patch is completely unrelated to whatever problem you are experiencing.

**Comment by jesus2099** [comment]
Yes I was reading that git-p4.py file and began to understand it had nothing to do with my use. 😉

Thanks for taking the time, letting me know, too!

**Comment by jesus2099** [comment]
~I have to find my pre-corrupt files... before checking them out with git...~
~I have a 2018 zip file of the repo, when it was still called _settings_.~

Oh great! I can actually just see files correctly in GitHub and copy code, then paste in the jsee file that I resave as UTF-8 with BOM.
